gensim/test/test_corpora_dictionary.py

```diff
@@ -189,32 +191,32 @@ def test_saveAsText(self):
         d = Dictionary(small_text)
 
         d.save_as_text(tmpf)
-        with open(tmpf) as file:
+        with codecs.open(tmpf, 'r', encoding='utf-8') as file:
             serialized_lines = file.readlines()
-            self.assertEqual(serialized_lines[0], "3\n")
+            self.assertEqual(serialized_lines[0], u"3\n")
             self.assertEqual(len(serialized_lines), 4)
             # We do not know, which word will have which index
-            self.assertEqual(serialized_lines[1][1:], "\tdruhé\t2\n")
-            self.assertEqual(serialized_lines[2][1:], "\tprvé\t1\n")
-            self.assertEqual(serialized_lines[3][1:], "\tslovo\t3\n")
+            self.assertEqual(serialized_lines[1][1:], u"\tdruhé\t2\n")
+            self.assertEqual(serialized_lines[2][1:], u"\tprvé\t1\n")
+            self.assertEqual(serialized_lines[3][1:], u"\tslovo\t3\n")
 
         d.save_as_text(tmpf, sort_by_word=False)
-        with open(tmpf) as file:
+        with codecs.open(tmpf, 'r', encoding='utf-8') as file:
```

Comment:
-1: use `smart_open`, binary mode `rb`, and convert to unicode explicitly (avoid `codecs`).